### DISCUSSION TITLE
Multipart uploading file by the PUT method

### DISCUSSION THREAD

**Discussion by abozhiy** [socketry/async-http#141] [Q&A] [answered]
I need to implement multipart upload file to S3 with presigned part URLs, for each part I need to pass file in request body by the PUT method, but as I understand it, async-http expects an array of strings for the body. Is it possible to pass a file inside the request body and how to do it? 
Thank you.

**Comment by abozhiy** [comment, marked as answer]
I think I've found solution:
`::Protocol::HTTP::Body::File.open(part_path)`

**Reply by olleolleolle** [reply]
Looks reasonable!
https://github.com/socketry/protocol-http/blob/main/lib/protocol/http/body/file.rb#L15-L17
